GOAT-Compiler/Code Generation/CNCMachine.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+
+namespace GOAT_Compiler.Code_Generation
+{
+    /// <summary>
+    /// The class that holds information about the CNC machine.
+    /// </summary>
+    public class CNCMachine
+    {
+        private double _currentExtrusion = 0;
+        private double _bedTemp = 0;
+        private double _extruderTemp = 0;
+        private double _fanPower = 0;
+
+        /// <summary>
+        /// The current positon of the extruder.
+        /// </summary>
+        public Vector Position { get; set; } = new Vector(0, 0, 0);
+
+        /// <summary>
+        /// The current extrusion.
```

Comment:
Plz give better summary, this is the current amount of extrusion, aka. distance*extrusionRate ?

Reply:
fixed